desc/compute/_profiles.py

```diff
@@ -0,0 +1,451 @@
+        current_term_r = (
+            mu_0
+            / (2 * jnp.pi)
+            * profiles["current"].compute(params["c_l"], dr=1)
+            / data["psi_r"]
+            - current_term * data["psi_rr"] / data["psi_r"]
+        )
+        num_r = (
+            data["lambda_rz"] * data["g_tt"]
+            + data["lambda_z"] * data["g_tt_r"]
+            - data["lambda_rt"] * data["g_tz"]
+            - (1 + data["lambda_t"]) * data["g_tz_r"]
+        ) / data["sqrt(g)"] - num * data["sqrt(g)_r"] / data["sqrt(g)"]
+        den_r = (
+            data["g_tt_r"] / data["sqrt(g)"]
+            - data["g_tt"] * data["sqrt(g)_r"] / data["sqrt(g)"] ** 2
+        )
+        num_avg_r = surface_averages(transforms["grid"], num_r)
+        den_avg_r = surface_averages(transforms["grid"], den_r)
+        data["iota_r"] = (
+            current_term_r + num_avg_r - data["iota"] * den_avg_r
+        ) / den_avg
+    return data
+
+
+# TODO: add K(rho,theta,zeta)*grad(rho) term
+@register_compute_fun(
+    name="I",
+    label="I",
+    units="T \\cdot m",
+    units_long="Tesla * meters",
+    description="Boozer toroidal current",
+    dim=1,
+    params=[],
+    transforms={"grid": []},
+    profiles=[],
+    function_of="r",
+    data=["B_theta"],
+)
+def _I(params, transforms, profiles, data, **kwargs):
+    data["I"] = surface_averages(transforms["grid"], data["B_theta"])
+    return data
+
+
+@register_compute_fun(
+    name="I_r",
+    label="\\partial_{\\rho} I",
+    units="T \\cdot m",
+    units_long="Tesla * meters",
+    description="Boozer toroidal current, derivative wrt radial coordinate",
+    dim=1,
+    params=[],
+    transforms={"grid": []},
+    profiles=[],
+    function_of="r",
+    data=["B_theta_r"],
+)
+def _I_r(params, transforms, profiles, data, **kwargs):
+    data["I_r"] = surface_averages(transforms["grid"], data["B_theta_r"])
+    return data
+
+
+@register_compute_fun(
+    name="G",
+    label="G",
+    units="T \\cdot m",
+    units_long="Tesla * meters",
+    description="Boozer poloidal current",
+    dim=1,
+    params=[],
+    transforms={"grid": []},
+    profiles=[],
+    function_of="r",
+    data=["B_zeta"],
+)
+def _G(params, transforms, profiles, data, **kwargs):
+    data["G"] = surface_averages(transforms["grid"], data["B_zeta"])
+    return data
+
+
+@register_compute_fun(
+    name="G_r",
+    label="\\partial_{\\rho} G",
+    units="T \\cdot m",
+    units_long="Tesla * meters",
+    description="Boozer poloidal current, derivative wrt radial coordinate",
+    dim=1,
+    params=[],
+    transforms={"grid": []},
+    profiles=[],
+    function_of="r",
+    data=["B_zeta_r"],
+)
+def _G_r(params, transforms, profiles, data, **kwargs):
+    data["G_r"] = surface_averages(transforms["grid"], data["B_zeta_r"])
+    return data
+
+
+@register_compute_fun(
+    name="current",
+    label="\\frac{2\\pi}{\\mu_0} I",
+    units="A",
+    units_long="Amperes",
+    description="Net toroidal current",
```

Comment:
could add "enclosed by flux surfaces" to be clear this is not just total net toroidal current